test-app/tests/integration/helpers/file-queue-helper-test.gts

```diff
@@ -274,4 +274,61 @@ module('Integration | Helper | file-queue', function (hooks) {
     await settled();
     assert.dom('[data-test-file]').doesNotExist('queue was flushed');
   });
+
+  // Service before helper issue ref: https://github.com/adopted-ember-addons/ember-file-upload/issues/1085
+  test('service can be used before helper with DEFUALT_QUEUE', async function (assert) {
+    const fileQueueService = this.owner.lookup(
+      'service:file-queue',
+    ) as FileQueueService;
+
+    await render(
+      <template>
+        <ul>
+          {{#each fileQueueService.files as |file|}}
+            <li>
+              {{file.name}}
+            </li>
+          {{/each}}
+        </ul>
+
+        {{#let (fileQueue) as |queue|}}
+          <label>
+            <input type='file' {{queue.selectFile}} hidden />
+            Select File
+          </label>
+        {{/let}}
+      </template>,
+    );
+
+    assert.dom('label').exists('regression would throw during rendering');
+  });
+
+  test('service can be used before helper with named queue, provided FileQueueService.create is called first', async function (assert) {
+    const fileQueueService = this.owner.lookup(
+      'service:file-queue',
+    ) as FileQueueService;
+    const customName = 'uploads';
+    fileQueueService.create(customName);
```

Comment:
In the case of a named queue. A user would need to manually create it before a rendering pass to avoid this issue.